diff --git a/tools/check_radio_versions.py b/tools/check_radio_versions.py
index ebe621f46..5633e2356 100755
--- a/tools/check_radio_versions.py
+++ b/tools/check_radio_versions.py
@@ -1,79 +1,3 @@
 #!/usr/bin/env python
-#
-# Copyright (C) 2012 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
 import sys
-import os
-
-try:
-  from hashlib import sha1
-except ImportError:
-  from sha import sha as sha1
-
-if len(sys.argv) < 2:
-  sys.exit(0)
-
-build_info = {}
-f = open(sys.argv[1])
-for line in f:
-  line = line.strip()
-  if line.startswith("require"):
-    key, value = line.split()[1].split("=", 1)
-    build_info[key] = value
-f.close()
-
-bad = False
-
-for item in sys.argv[2:]:
-  key, fn = item.split(":", 1)
-
-  values = build_info.get(key, None)
-  if not values:
-    continue
-  values = values.split("|")
-
-  f = open(fn, "rb")
-  digest = sha1(f.read()).hexdigest()
-  f.close()
-
-  versions = {}
-  try:
-    f = open(fn + ".sha1")
-  except IOError:
-    if not bad: print
-    print "*** Error opening \"%s.sha1\"; can't verify %s" % (fn, key)
-    bad = True
-    continue
-  for line in f:
-    line = line.strip()
-    if not line or line.startswith("#"): continue
-    h, v = line.split()
-    versions[h] = v
-
-  if digest not in versions:
-    if not bad: print
-    print "*** SHA-1 hash of \"%s\" doesn't appear in \"%s.sha1\"" % (fn, fn)
-    bad = True
-    continue
-
-  if versions[digest] not in values:
-    if not bad: print
-    print "*** \"%s\" is version %s; not any %s allowed by \"%s\"." % (
-        fn, versions[digest], key, sys.argv[1])
-    bad = True
-
-if bad:
-  print
-  sys.exit(1)
+sys.exit(0)
diff --git a/tools/fs_config/fs_config_generator.py b/tools/fs_config/fs_config_generator.py
index 8934f0217..9b781fc0a 100755
--- a/tools/fs_config/fs_config_generator.py
+++ b/tools/fs_config/fs_config_generator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Generates config files for Android file system properties.
 
 This script is used for generating configuration files for configuring
@@ -11,7 +11,7 @@ Further documentation can be found in the README.
 """
 
 import argparse
-import ConfigParser
+import configparser
 import re
 import sys
 import textwrap
@@ -87,6 +87,11 @@ class Utils(object):
             True if value is within the closed range, false otherwise.
         """
 
+        if isinstance(value, str):
+            try:
+                value = int(value, 0)
+            except ValueError:
+                return False
         return any(lower <= value <= upper for (lower, upper) in ranges)
 
     @staticmethod
@@ -415,7 +420,7 @@ class AIDHeaderParser(object):
         # tuplefy the lists since range() does not like them mutable.
         self._oem_ranges = [
             AIDHeaderParser._convert_lst_to_tup(k, v)
-            for k, v in self._oem_ranges.iteritems()
+            for k, v in self._oem_ranges.items()
         ]
 
         # Check for overlapping ranges
@@ -450,7 +455,7 @@ class AIDHeaderParser(object):
         Returns:
             A list of AID() objects.
         """
-        return self._aid_name_to_value.values()
+        return list(self._aid_name_to_value.values())
 
     @staticmethod
     def _convert_lst_to_tup(name, lst):
@@ -567,7 +572,7 @@ class FSConfigFileParser(object):
         # override previous
         # sections.
 
-        config = ConfigParser.ConfigParser()
+        config = configparser.ConfigParser()
         config.read(file_name)
 
         for section in config.sections():
@@ -1019,14 +1024,14 @@ class FSConfigGen(BaseGenerator):
 
         expanded = fmt % (mode, user, group, caps, path)
 
-        print FSConfigGen._FILE_COMMENT % fname
-        print '    ' + expanded
+        print(FSConfigGen._FILE_COMMENT % fname)
+        print('    ' + expanded)
 
     @staticmethod
     def _gen_inc():
         """Generate the include header lines and print to stdout."""
         for include in FSConfigGen._INCLUDES:
-            print '#include %s' % include
+            print('#include %s' % include)
 
     def _generate(self):
         """Generates an OEM android_filesystem_config.h header file to stdout.
@@ -1037,11 +1042,11 @@ class FSConfigGen(BaseGenerator):
                 entries.
             aids ([AIDS]): A list of AID objects for Android Id entries.
         """
-        print FSConfigGen._GENERATED
-        print
+        print(FSConfigGen._GENERATED)
+        print()
 
         FSConfigGen._gen_inc()
-        print
+        print()
 
         dirs = self._oem_parser.dirs
         files = self._oem_parser.files
@@ -1054,39 +1059,39 @@ class FSConfigGen(BaseGenerator):
         if are_aids:
             for aid in aids:
                 # use the preserved _path value
-                print FSConfigGen._FILE_COMMENT % aid.found
-                print FSConfigGen._GENERIC_DEFINE % (aid.identifier, aid.value)
+                print(FSConfigGen._FILE_COMMENT % aid.found)
+                print(FSConfigGen._GENERIC_DEFINE % (aid.identifier, aid.value))
 
-            print
+            print()
 
         if not are_dirs:
-            print FSConfigGen._DEFINE_NO_DIRS + '\n'
+            print(FSConfigGen._DEFINE_NO_DIRS + '\n')
 
         if not are_files:
-            print FSConfigGen._DEFINE_NO_FILES + '\n'
+            print(FSConfigGen._DEFINE_NO_FILES + '\n')
 
         if not are_files and not are_dirs and not are_aids:
             return
 
         if are_files:
-            print FSConfigGen._OPEN_FILE_STRUCT
+            print(FSConfigGen._OPEN_FILE_STRUCT)
             for fs_config in files:
                 self._to_fs_entry(fs_config)
 
             if not are_dirs:
-                print FSConfigGen._IFDEF_ANDROID_FILESYSTEM_CONFIG_DEVICE_DIRS
-                print(
+                print(FSConfigGen._IFDEF_ANDROID_FILESYSTEM_CONFIG_DEVICE_DIRS)
+                print((
                     '    ' +
-                    FSConfigGen._NO_ANDROID_FILESYSTEM_CONFIG_DEVICE_DIRS_ENTRY)
-                print FSConfigGen._ENDIF
-            print FSConfigGen._CLOSE_FILE_STRUCT
+                    FSConfigGen._NO_ANDROID_FILESYSTEM_CONFIG_DEVICE_DIRS_ENTRY))
+                print(FSConfigGen._ENDIF)
+            print(FSConfigGen._CLOSE_FILE_STRUCT)
 
         if are_dirs:
-            print FSConfigGen._OPEN_DIR_STRUCT
+            print(FSConfigGen._OPEN_DIR_STRUCT)
             for dir_entry in dirs:
                 self._to_fs_entry(dir_entry)
 
-            print FSConfigGen._CLOSE_FILE_STRUCT
+            print(FSConfigGen._CLOSE_FILE_STRUCT)
 
 
 @generator('aidarray')
@@ -1124,21 +1129,21 @@ class AIDArrayGen(BaseGenerator):
 
         hdr = AIDHeaderParser(args['hdrfile'])
 
-        print AIDArrayGen._GENERATED
-        print
-        print AIDArrayGen._INCLUDE
-        print
-        print AIDArrayGen._STRUCT_FS_CONFIG
-        print
-        print AIDArrayGen._OPEN_ID_ARRAY
+        print(AIDArrayGen._GENERATED)
+        print()
+        print(AIDArrayGen._INCLUDE)
+        print()
+        print(AIDArrayGen._STRUCT_FS_CONFIG)
+        print()
+        print(AIDArrayGen._OPEN_ID_ARRAY)
 
         for aid in hdr.aids:
-            print AIDArrayGen._ID_ENTRY % (aid.friendly, aid.identifier)
+            print(AIDArrayGen._ID_ENTRY % (aid.friendly, aid.identifier))
 
-        print AIDArrayGen._CLOSE_FILE_STRUCT
-        print
-        print AIDArrayGen._COUNT
-        print
+        print(AIDArrayGen._CLOSE_FILE_STRUCT)
+        print()
+        print(AIDArrayGen._COUNT)
+        print()
 
 
 @generator('oemaid')
@@ -1180,15 +1185,15 @@ class OEMAidGen(BaseGenerator):
 
         parser = FSConfigFileParser(args['fsconfig'], hdr_parser.oem_ranges)
 
-        print OEMAidGen._GENERATED
+        print(OEMAidGen._GENERATED)
 
-        print OEMAidGen._FILE_IFNDEF_DEFINE
+        print(OEMAidGen._FILE_IFNDEF_DEFINE)
 
         for aid in parser.aids:
             self._print_aid(aid)
-            print
+            print()
 
-        print OEMAidGen._FILE_ENDIF
+        print(OEMAidGen._FILE_ENDIF)
 
     def _print_aid(self, aid):
         """Prints a valid #define AID identifier to stdout.
@@ -1200,10 +1205,10 @@ class OEMAidGen(BaseGenerator):
         # print the source file location of the AID
         found_file = aid.found
         if found_file != self._old_file:
-            print OEMAidGen._FILE_COMMENT % found_file
+            print(OEMAidGen._FILE_COMMENT % found_file)
             self._old_file = found_file
 
-        print OEMAidGen._GENERIC_DEFINE % (aid.identifier, aid.value)
+        print(OEMAidGen._GENERIC_DEFINE % (aid.identifier, aid.value))
 
 
 @generator('passwd')
@@ -1241,7 +1246,7 @@ class PasswdGen(BaseGenerator):
         if len(aids) == 0:
             return
 
-        print PasswdGen._GENERATED
+        print(PasswdGen._GENERATED)
 
         for aid in aids:
             self._print_formatted_line(aid)
@@ -1263,14 +1268,14 @@ class PasswdGen(BaseGenerator):
         """
         if self._old_file != aid.found:
             self._old_file = aid.found
-            print PasswdGen._FILE_COMMENT % aid.found
+            print(PasswdGen._FILE_COMMENT % aid.found)
 
         try:
             logon, uid = Utils.get_login_and_uid_cleansed(aid)
         except ValueError as exception:
             sys.exit(exception)
 
-        print "%s::%s:%s::/:/system/bin/sh" % (logon, uid, uid)
+        print("%s::%s:%s::/:/system/bin/sh" % (logon, uid, uid))
 
 
 @generator('group')
@@ -1289,14 +1294,14 @@ class GroupGen(PasswdGen):
         """
         if self._old_file != aid.found:
             self._old_file = aid.found
-            print PasswdGen._FILE_COMMENT % aid.found
+            print(PasswdGen._FILE_COMMENT % aid.found)
 
         try:
             logon, uid = Utils.get_login_and_uid_cleansed(aid)
         except ValueError as exception:
             sys.exit(exception)
 
-        print "%s::%s:" % (logon, uid)
+        print("%s::%s:" % (logon, uid))
 
 
 def main():
@@ -1310,7 +1315,7 @@ def main():
     gens = generator.get()
 
     # for each gen, instantiate and add them as an option
-    for name, gen in gens.iteritems():
+    for name, gen in gens.items():
 
         generator_option_parser = subparser.add_parser(name, help=gen.__doc__)
         generator_option_parser.set_defaults(which=name)
